### PR TITLE
feat(Header): Add returnUrl for unauthenticated links

### PR DESCRIPTION
**Pull request by mengtzu**
Accepts optional string prop `returnUrl` on the `Header` component.  If present, appends it (URI encoded) to the href for the Sign in and Register links as a query string param `?returnUrl={returnUrl}`

### CODE REVIEW

File: react/Header/SignInRegister/SignInRegister.js [hunk 1]
```diff
@@ -4,7 +4,10 @@ import React, { PropTypes } from 'react';
 
 import ScreenReaderOnly from '../../ScreenReaderOnly/ScreenReaderOnly';
 
-export default function SignInRegister({ linkRenderer }) {
+export default function SignInRegister({ linkRenderer, returnUrl }) {
```
Comment (mattsjones) [inline react/Header/SignInRegister/SignInRegister.js]: Can you add a prop type for `returnUrl`

Comment (mengtzu) [inline react/Header/SignInRegister/SignInRegister.js]: Types are the cancer that is killing Javascript.

But sure :)

Comment (mengtzu) [inline react/Header/SignInRegister/SignInRegister.js]: Done

**Comment by markdalgleish** [comment]
@mengtzu can you add a snapshot test, or modify an existing one?

File: react/Header/Header.demo.js [hunk 2]
```diff
@@ -16,7 +16,8 @@ export default {
     authenticationStatus: 'authenticated',
     userName: 'Jane Smith',
     divider: false,
-    linkRenderer: dummyLinkRenderer
+    linkRenderer: dummyLinkRenderer,
+    returnUrl: 'seek-oss.github.io/seek-style-guide'
```
Comment (markdalgleish) [inline react/Header/Header.demo.js]: Super minor nitpick, but could you make this `/jobs`, since we don't support cross-domain return URLs?

**Comment by mengtzu** [comment]
@markdalgleish Added a snapshot, also added a helper script to update them.  My first Jest, let me know if it's wrong.

File: package.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
     "start": "BASE_HREF=/ node server",
     "test": "npm run test-unit && npm run lint && npm run build",
     "test-unit": "BABEL_ENV=test jest --config ./test/jest.config.json",
+    "update-snapshot": "BABEL_ENV=test jest --config ./test/jest.config.json --updateSnapshot",
```
Comment (markdalgleish) [inline package.json]: To ensure this stays consistent, it's probably worth changing this to `npm run test-unit -- --updateSnapshot`

Comment (mengtzu) [inline package.json]: done

**Pull request merged**